src/picasso-def/components/labels.js

```diff
@@ -222,19 +222,39 @@ export const displayInvalidMessage = ({ rect, text, renderer, theme }) => {
   ];
 };
 
-export const createOverlayLabel = ({ node, avgColor, width, height, theme }) => {
+export const createOverlayLabel = ({ node, avgColor, width, height, renderer, theme }) => {
+  let lines = [];
   const text = node.data.label;
+  const maxNumLines = /\s+/.test(text) ? 2 : 1; // allow break if white space in label
   const fontFamily = theme.getStyle('object.treemap', 'label.value', 'fontFamily') || 'Source Sans Pro';
   const optimal = Math.round(0.1 * Math.sqrt(2 * width * height));
   const fontSize = `${optimal}px`;
+
+  const textSize = renderer.measureText({
+    text,
+    fontSize,
+    fontFamily,
+  });
+
+  let finalText = '';
+  if (textSize.width > width - 8) {
+    lines = wrapText(text, width - 10, maxNumLines, ellipsis, renderer, fontSize, fontFamily);
+    if (Array.isArray(lines) && lines.length > 0) {
+      lines.forEach((line) => {
+        finalText = finalText + line + '\n';
+      });
+    }
+  } else {
+    finalText = text;
```

Comment:
in this case (where there is oly space for one line) you don't get any ellipsis when the text is to long

Reply:
The if condition with width is checking that and the scenario with one line and long text is handled there as can be seen in the [screenshot.
![Screenshot 2023-01-30 at 12 07 53](https://user-images.githubusercontent.com/30860406/215461098-b1b0a837-e255-4460-8f24-2ce99dc42b48.png)

Reply:
You are correct.
(a saw a problem, but that was when the height only allowed one line. and have to be somewhere else in the code)